bool[] occupancy_values
int64 rows
int64 columns